Modules/DSCParser/Modules/DSCParser.psm1

```diff
@@ -80,17 +84,70 @@ function ConvertFrom-CIMInstanceToHashtable
         $statements = $ChildObject.PipelineElements.Expression.SubExpression.Statements
         foreach ($statement in $statements)
         {
-            $result += ConvertFrom-CIMInstanceToHashtable -ChildObject $statement
+            $result += ConvertFrom-CIMInstanceToHashtable -ChildObject $statement `
+                                                          -ResourceName $ResourceName
         }
     }
     else
     {
         $result = @{}
         $KeyPairs = $ChildObject.CommandElements[2].KeyValuePairs
         $CIMInstanceName = $ChildObject.CommandElements[0].Value
+
+        # Get the CimClass associated with the current CimInstanceName
+        $CIMClassObject = Get-CimClass -ClassName $CimInstanceName `
```

Comment:
This call will silently fail if the user is not an administrator. So will all further calls in 122 and 130.

I would add some check like this

```powershell
if ((-not(([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)))     {
#         Print a message to recommend elevated usage for better parsing
        }